### PR TITLE
Preparation for node-controller-integration

### PR DESCRIPTION
**Pull request by mrIncompetent**

### CODE REVIEW

File: glide.yaml [hunk 1]
```diff
@@ -4,6 +4,7 @@ import:
   version: v3.0.0-beta.0
 - package: github.com/auth0/go-jwt-middleware
 - package: github.com/digitalocean/godo
+  version: 1.*
```
Comment (lbb) [inline glide.yaml]: please use ^1.0.0

File: cmd/kubermatic-cluster-controller/main.go [hunk 2]
```diff
@@ -1,7 +1,119 @@
...
+	versionsFile          = flag.String("versions", "versions.yaml", "The versions.yaml file path")
+	updatesFile           = flag.String("updates", "updates.yaml", "The updates.yaml file path")
+	apiserverExternalPort = flag.Int("apiserver-external-port", 8443, "Port on which the apiserver of a client cluster should be reachable")
+)
 
 func main() {
-	cmd.Execute()
+	flag.Parse()
+
+	if *masterResources == "" {
```
Comment (lbb) [inline cmd/kubermatic-cluster-controller/main.go:35]: also check `externalURL `(Required)

File: cmd/kubermatic-cluster-controller/main.go [hunk 3]
```diff
@@ -1,7 +1,119 @@
...
+
+	dcs, err := provider.DatacentersMeta(*dcFile)
+	if err != nil {
+		glog.Fatal(fmt.Printf("failed to load datacenter yaml %q: %v", *dcFile, err))
+	}
+
+	// load versions
+	versions, err := version.LoadVersions(*versionsFile)
+	if err != nil {
+		glog.Fatal(fmt.Printf("failed to load version yaml %q: %v", *versionsFile, err))
```
Comment (lbb) [inline cmd/kubermatic-cluster-controller/main.go]: Remove `fmt.Printf` or change it to `fmt.Sprintf`

File: cmd/kubermatic-cluster-controller/main.go [hunk 4]
```diff
@@ -1,7 +1,119 @@
...
+	}
+
+	// load updates
+	updates := []api.MasterUpdate{}
+	if *updatesFile != "" {
+		var err error
+		updates, err = version.LoadUpdates(*updatesFile)
+		if err != nil {
+			glog.Fatal(fmt.Printf("failed to load updates yaml %q: %v", *updatesFile, err))
```
Comment (lbb) [inline cmd/kubermatic-cluster-controller/main.go]: Remove `fmt.Printf` or change it to `fmt.Sprintf`

**Review by lbb — CHANGES_REQUESTED** [review]

File: cmd/kubermatic-cluster-controller/main.go (hunk 2 above)
Comment (mrIncompetent) [reply cmd/kubermatic-cluster-controller/main.go:35]: fixed

File: cmd/kubermatic-api/main.go [hunk 5]
```diff
@@ -1,7 +1,67 @@
...
+
+var (
+	workerName       = flag.String("worker-name", "", "Create clusters only processed by worker-name cluster controller")
+	kubeConfig       = flag.String("kubeconfig", "", "The kubeconfig file path with one context per Kubernetes provider")
+	auth             = flag.Bool("auth", true, "Activate authentication with JSON Web Tokens")
+	dcFile           = flag.String("datacenters", "datacenters.yaml", "The datacenters.yaml file path")
+	jwtKey           = flag.String("jwt-key", "", "The JSON Web Token validation key, encoded in base64")
+	address          = flag.String("address", ":8080", "The address to listen on")
+	masterKubeconfig = flag.String("master-kubeconfig", "", "When set it will overwrite the usage of the InClusterConfig")
+)
```
Comment (scheeles) [inline cmd/kubermatic-api/main.go:32]: secretsFile is not required any more or?

**Review by scheeles — APPROVED** [review]
LGTM

File: cmd/kubermatic-api/main.go (hunk 5 above)
Comment (mrIncompetent) [reply cmd/kubermatic-api/main.go:32]: Nope, i removed the code which required it. We never used or even maintained it

**Pull request merged**